test/lib/testAttributeEditorTemplate.py

```diff
@@ -0,0 +1,255 @@
+#!/usr/bin/env python
+
+#
+# Copyright 2023 Autodesk
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import unittest
+
+import mayaUtils
+import ufeUtils
+import fixturesUtils
+import testUtils
+
+import mayaUsd.lib
+
+from maya import cmds
+from maya import standalone
+
+from mayaUsdLibRegisterStrings import getMayaUsdLibString
+
+from pxr import Usd
+
+import ufe
+
+class AttributeEditorTemplateTestCase(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        fixturesUtils.readOnlySetUpClass(__file__, initializeStandalone=False)
+
+    def setUp(self):
+        cmds.file(new=True, force=True)
+        cmds.select(clear=True)
+
+    def searchForMayaControl(self, controlOrLayout, mayaControlCmd, labelToFind):
+        '''Helper function to search for a Maya control or layout with the label
+        matching the input string <p labelToFind>. The type of the control being
+        searched for is also given as input <p mayaControlCmd>.
+        '''
+        if mayaControlCmd(controlOrLayout, exists=True):
+            if mayaControlCmd(controlOrLayout, q=True, label=True) == labelToFind:
+                return controlOrLayout
+
+        if controlOrLayout:
+            childrenOfLayout = cmds.layout(controlOrLayout, q=True, ca=True)
+            if childrenOfLayout:
+                for child in childrenOfLayout:
+                    child = controlOrLayout + '|' + child
+                    if cmds.layout(child, exists=True):
+                        foundControl = self.searchForMayaControl(child, mayaControlCmd, labelToFind)
+                        if foundControl:
+                            return foundControl
+                    elif mayaControlCmd(child, exists=True):
+                        if mayaControlCmd(child, q=True, label=True) == labelToFind:
+                            return child
+        return None
```

Comment:
This function searched thru the layouts in the AE for either a layout (in this case we are searching for frame layout which is used for each section) with a specific name or a control (of a certain type) with a label equal to input name.